lib/apiservers/portlayer/swagger.json

```diff
@@ -175,8 +176,166 @@
+				      "tags": [
+					        "storage"
+				      ],
+				      "consumes": [
+					        "application/json"
+				      ],
+				      "produces": [
+					        "application/x-tar"
+				      ],
+				      "operationId": "ExportArchive",
+				      "parameters": [
+						{
+							"name": "store",
+							"in":"query",
+							"required":true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name": "ancestorStore",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"deviceID",
+							"in":"query",
+							"required": true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"ancestor",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"data",
+							"in":"query",
+							"required": true,
+							"type" : "boolean",
+							"x-nullable": false						
+						},
+						{
+							"name":"filterSpec",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": true
+						}
+					],
+					"responses": {
+						"200": {
+							"description": "OK",
+							"schema": {
+								"format": "binary"
+							}
+						},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"423": {
+						"description": "Device or resource is locked",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"500": {
+						"description": "failed to export tar archive from target"
+					},
+					"default": {
+						"description": "error",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					}
+				}
+          },
+      "post": {
+          "description": "Imports a tar stream to a device at the target path",
+				  "summary": "Import tar stream to supplied target",
+				  "tags": [
+					        "storage"
+				  ],
+				  "consumes": [
+					    "application/x-tar"
+				  ],
+				  "operationId": "ImportArchive",
+				  "parameters": [
+					{
+						"name": "store",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"deviceID",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"filterSpec",
+						"in":"query",
+						"required": false,
+						"type" : "string",
+						"x-nullable": true
+					},
+					{
+						"name": "Archive",
+						"required": true,
+						"in": "body",
+						"schema": {
+							"type":"string",
+							"format":"binary"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
```

Comment:
For the time being, just return the http status without the modeled errors.  That's another project that I'll add for the future, which we'll use in all of the streaming code.  We currently do not return any modeled errors in any of our streaming code right now.